pb/metainfo.proto

```diff
@@ -781,10 +786,45 @@ message ObjectFinishMoveRequest {
     repeated EncryptedKeyAndNonce new_segment_keys = 5;
 }
 
+message ObjectFinishMoveResponse { }
+
+// Server side copy.
+// Has the same fields as move.
+message ObjectBeginCopyRequest {
+    RequestHeader header = 15;
+
+    bytes source_bucket = 1;
+    bytes source_encrypted_object_key = 2;
+
+    bytes destination_bucket = 3;
+    bytes destination_encrypted_object_key = 4;
+}
+
+message ObjectBeginCopyResponse {
+    bytes destination_stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+
+    bytes source_encrypted_metadata_key_nonce = 2 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes source_encrypted_metadata_key = 3;
+    repeated EncryptedKeyAndNonce source_segment_keys = 4;
+    encryption.EncryptionParameters source_encryption_parameters = 5;
+}
+
+message ObjectFinishCopyRequest {
+    RequestHeader header = 15;
+
+    bytes destination_stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
```

Comment:
I think this will be also source_stream_id as destination stream id will be generated inside finish copy operation and won't be exposed. At least in initial version.